src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -263,6 +241,38 @@ await listener.AfterExecutionAsync(context)
             return result;
         }
 
+        /// <summary>
+        /// Builds a <see cref="ExecutionContext"/> instance from the provided values.
+        /// </summary>
+        protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions options, Document document, Operation operation, Variables variables, Metrics metrics)
+        {
+            var context = new ExecutionContext
+            {
+                Document = document,
+                Schema = options.Schema,
+                RootValue = options.Root,
+                UserContext = options.UserContext,
+
+                Operation = operation,
+                Variables = variables,
+                Fragments = document.Fragments,
+                Errors = new ExecutionErrors(),
+                Extensions = new Dictionary<string, object>(),
```

Comment:
Also I suggest to remove this initialization line. Anyone who needs extensions can now override `BuildExecutionContext` ans set it.

Reply:
I'll separately do a PR for this after this PR is merged.